src/transactions/TransactionFrame.cpp

```diff
@@ -148,18 +148,26 @@ TransactionFrame::getMinFee(LedgerHeader const& header) const
 }
 
 int64_t
-TransactionFrame::getFee(LedgerHeader const& header, int64_t baseFee) const
+TransactionFrame::getFee(LedgerHeader const& header, int64_t baseFee,
+                         bool applying) const
 {
-    if (header.ledgerVersion < 11)
-    {
-        return getFeeBid();
-    }
-    else
+    if (header.ledgerVersion >= 11 || !applying)
```

Comment:
I'm having a hard time understanding this. I _think_ this works even before version 11 because in that case `baseFee = lh.baseFee`. I also _think_ this works correctly for fee-bump inner transactions which will have `baseFee = 0`. It's pretty hard to see on preliminary inspection though. Can you add some commentary explaining why this is right?

I'm also not sure why this matters, given that no fee recommendations from before version 11 can actually be accessed (I suppose this might happen on the testnet during a reset?).

Reply:
I'm not 100% sure I understand your question.

There are two changes here:
* use the same code path when validating regardless of protocol version (as validation is not part of the protocol, I'd rather take the simpler route). This is just to simplify testing and not have to special case this in SDKs/Horizon
* not clamping down during validation. I think this is necessary because it's possible to have `adjustedFee > getFeeBid()` because the effective base fee (during surge pricing) is computed with a round up and I didn't want to potentially give a bid that was potentially too small